_episodes/03-high-dim-modeling.md

```diff
@@ -4,43 +4,121 @@ teaching: 0
+
+## **The curse of dimensionality**
+- The phrase, attributed to Richard Bellman, was coined to express the difficulty of using brute force (a.k.a. grid search) to optimize a function with too many input variables.
+- In machine learning, the Curse of Dimensionality refers to a set of problems that arise when working with high-dimensional data.
+  - Compute Time: Generally speaking, as the number of features in a model increases, so does the amount of time required to opitmize the model
+  - Overfitting: When there are more features than training observations, overfitting is gauranteed
+  - Interpretability: Models become more difficult to interpret as the numbers of features increase
+  - Clustering: Too many dimensions causes every observation in your dataset to appear equidistant from all the others — making it difficult to form clusters on the data. If the distances are all approximately equal, then all the observations appear equally alike (as well as equally different), and no meaningful clusters can be formed.
+
+### **Bias-Variance Tradeoff (Underftting vs Overfitting)**
+- Train error:
+- Test error: 
+- Overfitting: 
+
+**Bias**: Bias is the model's average prediction error (i.e. model's predicted values minus actual values). A model with high bias pays very little attention to the training data and oversimplifies the model. It leads to a high error on both the train and test datasets.
+
+**Variance**: Variance is the variability of the model's predictions. It indicates the overal spread of the model's predictions. A model with high variance pays too much attention to the training data (i.e. overfits) 
+
+10X as many datapoints as features as general rule of thumb
+![image alt text](https://upload.wikimedia.org/wikipedia/commons/9/9f/Bias_and_variance_contributing_to_total_error.svg)
+![need to find equivalent image w/ creative commons license](https://www.cs.cornell.edu/courses/cs4780/2018fa/lectures/images/bias_variance/bullseye.png)
+
+**Important considerations / takeaways**
+
+
+# **Feature Selection**
+In machine learning and statistics, feature selecture (a.k.a. variable selection, attribute selection, or variable subset selection) is the process of selecting the best subset of relevant features for use in a given model/estimator. There are several benefits to incorporating feature selection into a machine learning analysis:
+1. simplification of models to make them easier to interpret by researchers/users
+2. shorter training times
+3. to avoid overfitting and subsequently improve the accuracy of the predictive models
+
+## Illustrate feature selection with simple exercise
+- use apriori knowledge to filter out unwanted features (e.g. remove constants).  
```

Comment:
might suggest changing apriori to plain language, or including both apriori and a plain lang definition